usb_protocol/emitters/descriptors/hid.py

```diff
@@ -0,0 +1,39 @@
+import unittest
+
+from contextlib import contextmanager
+
+from ..           import emitter_for_format
+from ..descriptor import ComplexDescriptorEmitter
+from ...types.descriptors.hid import *
+
+class HIDDescriptorEmitter(ComplexDescriptorEmitter):
+
+    DESCRIPTOR_FORMAT = HIDDescriptor
+    
+    @contextmanager
+    def ReportDescriptor(self):
+        """ Context manager that allows addition of a subordinate report descriptor.
+
+        It can be used with a `with` statement; and yields an HIDReportDescriptorEmitter
+        that can be populated:
+
+            with hiddescriptor.ReportDescriptor() as r:
+                r.wDescriptorLength = 0x10
+
+        This adds the relevant descriptor, automatically.
+        """
+
+        descriptor = HIDReportDescriptorEmitter()
+        yield descriptor
```

Comment:
Similarly, I think `ReportDescriptor` should be changed to `DescriptorReference` throughout these lines.

Reply:
Good suggestion, applied it